### DISCUSSION TITLE
Is this a valid base query?

### DISCUSSION THREAD

**Discussion by Roosteridk** [deta/cloud-docs#650] [Help] [answered]
Basically trying to do `SELECT * FROM base WHERE (key != 'a' && key != 'b')`

{
  "key?ne": 'a',
  "key?ne": 'b',
}

I know we can apply de morgan's law (in wihch case it would be the same as `!(key == a || key == b)`)  but I was curious if this worked -- just seems more natural.

**Comment by lemonyte** [comment, marked as answer]
> [When querying keys] the [AND](https://docs.deta.sh/docs/base/queries#and) and [OR](https://docs.deta.sh/docs/base/queries#or) operations for different query values are not supported

https://docs.deta.sh/docs/base/queries#querying-keys

```
{
    // different AND key queries (invalid query)
    "key": "a",
    "key?pfx": "b"
}
```

**Reply by Roosteridk** [reply]
Just figured it out, I gotta use range for the problem
